### DISCUSSION TITLE
Error: PyVISA is required for loading VISA instruments

### DISCUSSION THREAD

**Discussion by Schreiberling** [Galvant/InstrumentKit#367] [Q&A] [answered]
Hi,

I want to use your library to control and access a Yokogawa optical spectrum analyzer. However, when I try to open the device using open_visa (_ik.yokogawa.Yokogawa6370.open_visa('TCPIP0:192.168.0.35')_) I get following error:
```
Traceback (most recent call last):
  File "C:\....\lib\site-packages\instruments\abstract_instruments\instrument.py", line 605, in open_visa
    raise ImportError("PyVISA is required for loading VISA ")
ImportError: PyVISA is required for loading VISA instruments.
```

PyVISA is installed and when I use pyvisa directly according to their example everything works fine and I could also start a sweep.
```
import pyvisa
rm = pyvisa.ResourceManager()
rm.list_resources()
my_instrument = rm.open_resource('TCPIP0:192.168.0.35')
```

I found out that in the mentioned file _instruments.py_ 'import visa' is made instead of 'import pyvisa'. Could this be a problem? What else could it be?

I hope you can help me. Thank you!

EDIT: I just recognized that the file _instrument.py_ from the basic intrument class which I have on my copmuter is quite different comparedd to the code [here](https://instrumentkit.readthedocs.io/en/latest/_modules/instruments/abstract_instruments/instrument.html). Shouldn't those be the same? How can this be? I have installed the latest version.

**Comment by scasagrande** [comment, marked as answer]
`pip install -U instrumentkit --pre` or `pip install instrumentkit==1.0.0b1`
